tests/unit/test_charm.py

```diff
@@ -915,3 +915,214 @@ def test_postgres_patch_storage_config_defaults_to_database_relation(self):
         }
         environment = plan.to_dict()["services"]["livepatch"]["environment"]
         self.assertEqual(environment, environment | required_environment)
+
+    def test_pro_airgapped_server_relation__success(self):
+        """Test pro-airgapped-server relation."""
+        self.harness.set_leader(True)
+        self.harness.enable_hooks()
+
+        self.harness.charm._state.dsn = "postgresql://123"
+        self.harness.charm._state.resource_token = TEST_TOKEN
+
+        with patch("src.charm.LivepatchCharm.migration_is_required") as migration:
+            migration.return_value = False
+            self.harness.update_config(
+                {
+                    "server.url-template": "http://localhost/{filename}",
+                    "server.is-hosted": False,
+                }
+            )
+
+            pro_rel_id = self.harness.add_relation("pro-airgapped-server", "pro-airgapped-server")
+            self.harness.add_relation_unit(pro_rel_id, "pro-airgapped-server/0")
+            self.harness.update_relation_data(
+                pro_rel_id,
+                "pro-airgapped-server/0",
+                {
+                    "scheme": "scheme",
+                    "hostname": "some.host.name",
```

Comment:
I would rather go for `"http"` scheme and some random private IP :) just to make tests look more real.

Reply:
This is intentional, to make sure the charm doesn't assume anything and uses the provided scheme as is.